### PR TITLE
pcsclite, pcsc{-d,tools}: fix cross, {systemd,udev,dbus}Support flags

### PR DESCRIPTION
**Pull request by flokli**
## Description of changes
Cross fixes and `{systemd,udev,dbus}Support` for `pcsclite, pcsc{-d,tools}`.

Sent by amjoseph at https://lists.sr.ht/~andir/nixpkgs-dev.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] pkgsCross.aarch64-multiplatform
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/tools/security/pcsc-tools/default.nix [hunk 1]
```diff
@@ -7,17 +7,22 @@
 , makeWrapper
 , pkg-config
 , wrapGAppsHook
-, systemd
-, dbus
+, systemdSupport ? stdenv.isLinux, systemd
```
Comment (NickCao) [inline pkgs/tools/security/pcsc-tools/default.nix:10]:
```suggestion
, systemdSupport ? lib.meta.availableOn stdenv.hostPlatform systemd, systemd
```

Comment (flokli) [reply pkgs/tools/security/pcsc-tools/default.nix:10]: Applied, thanks!

**Comment by flokli** [comment]
Successfully built (from x86_64-linux):

 - `pcsc-tools`
 - `pkgsCross.aarch64-multiplatform.pcsc-tools`
 - `pkgsCross.aarch64-multiplatform.pcsc-tools.override { systemdSupport = false; dbusSupport = false;}`
 - `pcsc-tools.override { systemdSupport = false; dbusSupport = false;}`

File: pkgs/tools/security/pcsc-tools/default.nix [hunk 2]
```diff
@@ -7,17 +7,22 @@
 , makeWrapper
 , pkg-config
 , wrapGAppsHook
-, systemd
-, dbus
+, systemdSupport ? lib.meta.availableOn stdenv.hostPlatform systemd, systemd
+, dbusSupport ? stdenv.isLinux, dbus
```
Comment (NickCao) [inline pkgs/tools/security/pcsc-tools/default.nix:11]: `lib.meta.availableOn` also applies here, and within pcsclite.

File: pkgs/tools/security/pcsclite/default.nix [hunk 3]
```diff
@@ -10,13 +10,20 @@
 , dbus
 , polkit
 , systemdLibs
+, dbusSupport ? stdenv.isLinux
+, systemdSupport ? stdenv.isLinux
+, udevSupport ? dbusSupport
```
Comment (NickCao) [inline pkgs/tools/security/pcsclite/default.nix:15]: Also check for availability?

File: pkgs/tools/security/pcsc-tools/default.nix (hunk 2 above)
Comment (flokli) [reply pkgs/tools/security/pcsc-tools/default.nix:11]: I don't think so. dbus is available on Darwin, but that doesn't mean it makes sense to build it for that platform. Though "if on Linux, enable dbus by default" is a good assumption.

File: pkgs/tools/security/pcsclite/default.nix (hunk 3 above)
Comment (flokli) [reply pkgs/tools/security/pcsclite/default.nix:15]: For systemd, yes. For dbus, see above.

Comment (flokli) [reply pkgs/tools/security/pcsclite/default.nix:15]: Done.

**Pull request merged**